_monitoring-plugins/ad/api.md

```diff
@@ -668,1043 +1929,1016 @@ POST _plugins/_anomaly_detection/detectors/results/_search
+        "end_time": 1633044725885
       },
-      "last_update_time": 1611716538071,
       "user": {
         "name": "admin",
         "backend_roles": [
           "admin"
         ],
         "roles": [
-          "all_access",
-          "own_index"
+          "own_index",
+          "all_access"
         ],
         "custom_attribute_names": [],
         "user_requested_tenant": "__user__"
-      },
-      "detector_type": "HISTORICAL_SINGLE_ENTITY",
-      "detection_date_range": {
-        "start_time": 1580094137997,
-        "end_time": 1611716537997
       }
     },
-    "user": {
-      "name": "admin",
-      "backend_roles": [
-        "admin"
-      ],
-      "roles": [
-        "all_access",
-        "own_index"
-      ],
-      "custom_attribute_names": [],
-      "user_requested_tenant": "__user__"
-    }
+    "shingle_size": 8,
+    "rcf_total_updates": 1994,
+    "threshold_model_trained": true,
+    "threshold_model_training_data_size": 0,
+    "model_size_in_bytes": 1593240,
+    "node_id": "dIyavWhmSYWGz65b4u-lpQ",
+    "detector_task_slots": 1
   }
 }
 ```
 
----
+The `total_entities` parameter shows you the total number of entities including the number of category fields for a detector.
 
-## Search detector
-Introduced 1.0
-{: .label .label-purple }
+Getting the total count of entities is an expensive operation for real-time analysis of a detector with more than one category field. By default, for a real-time detection profile, a detector counts the number of entities up to a value of 10,000. For historical analysis, the anomaly detection plugin only detects the top 1,000 entities by default and caches the top entities in memory, so it doesn't cost much to get the total count of entities for historical analysis.
 
-Returns all anomaly detectors for a search query.
+The `profile` operation also provides information about each entity, such as the entity’s `last_sample_timestamp` and `last_active_timestamp`. `last_sample_timestamp` shows the last document in the input data source index containing the entity, while `last_active_timestamp` shows the timestamp when the entity’s model was last seen in the model cache.
+
+If there are no anomaly results for an entity, either the entity doesn't have any sample data or its model is removed from the model cache.
 
 #### Request
 
 ```json
-GET _plugins/_anomaly_detection/detectors/_search
-POST _plugins/_anomaly_detection/detectors/_search
-
-Sample Input:
+GET _plugins/_anomaly_detection/detectors/<detectorId>/_profile?_all=true
 {
-  "query": {
-    "match": {
-      "name": "test-detector"
+  "entity": [
+    {
+      "name": "host",
+      "value": "i-00f28ec1eb8997686"
     }
-  }
+  ]
 }
 ```
 
-
-#### Sample response
+#### Sample Responses
 
 ```json
 {
-  "took": 13,
-  "timed_out": false,
-  "_shards": {
-    "total": 5,
-    "successful": 5,
-    "skipped": 0,
-    "failed": 0
+  "category_field": "host",
+  "value": "i-00f28ec1eb8997686",
```

Comment:
These two lines are removed in 1.1 as we have already provided name and value in the request body.